### PR TITLE
Focal search dynoplan

### PR DESCRIPTION
**Pull request by akmaralAW**
tdbA*_epsilon implementation based on libMultiRobotPlanning/astar_epsilon.

### CODE REVIEW

**Review by whoenig — APPROVED** [review]
Perhaps we can have a pair prog session where we diff tdbastar and tdbastar_epsilon? I find it difficult to review as it is a lot of code, but the changes between the two should be pretty minimal.
Also, it would make sense to add a test in the future. A simple one is that the output should be exactly the same if w=1.0. For larger w, solutions should be found faster (well, assuming the heuristic computation is fast enough).

File: src/tdbastar/tdbastar_epsilon.cpp [hunk 1]
```diff
@@ -0,0 +1,727 @@
...
+          if (t >= solution[robot_idx].trajectory.states.size()){
+            node_state = solution[robot_idx].trajectory.states.back();    
+          }
+          else {
+            node_state = solution[robot_idx].trajectory.states[t];
+          }
+          std::vector<fcl::Transform3d> tmp_ts(1);
+          if (robot->name == "car_with_trailers") {
+            tmp_ts.resize(2);
+          }
```
[the remark this answers is not in the document]
Comment (akmaralAW) [reply src/tdbastar/tdbastar_epsilon.cpp:91]: This is needed for collision checking. Now, in Quim's implementation, he gives std::vector<fcl::Transform3d> for           robot->transformation_collision_geometries. And in his implementation car (with/without trailer) expects you give the correct size vector for this. That is why I have vector of size 2 for the car with trailer.

File: src/tdbastar/tdbastar_epsilon.cpp [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
+    const std::vector<std::shared_ptr<dynobench::Model_robot>>& all_robots,
+    std::shared_ptr<fcl::BroadPhaseCollisionManagerd> col_mng_robots,
+    std::vector<fcl::CollisionObjectd*>& robot_objs){
+    size_t max_t = 0;
+    int numConflicts = 0;
+    // for the state we are checking
+    std::vector<fcl::Transform3d> current_tmp_ts(1);
+    if (all_robots[current_robot_idx]->name == "car_with_trailers") {
+      current_tmp_ts.resize(2);
+    }
```
[the remark this answers is not in the document]
Comment (akmaralAW) [reply src/tdbastar/tdbastar_epsilon.cpp:126]: the same as above

**Comment by quimortiz** [comment]
before doing the pull request in main, can you try to unify tdbastar and tdbastar epsilon in a single function?

**Comment by quimortiz** [comment]
+ add tests

**Pull request merged**